include/private/neubot/dash_impl.hpp

```diff
@@ -0,0 +1,547 @@
+                            ctx->cb(error);
+                            return;
+                        }
+                        assert(!!res);
+                        if (res->status_code != 200) {
+                            ctx->logger->warn("dash: invalid response code: %s",
+                                              error.explain().c_str());
+                            ctx->cb(http::HttpRequestFailedError());
+                            return;
+                        }
+                        res->request = req;
+                        auto length = res->body.length();
+                        double time_elapsed = mk::time_now() - saved_time;
+                        if (time_elapsed <= 0) { // For robustness
+                            ctx->logger->warn("dash: negative time error");
+                            ctx->cb(GenericError("negative_time_error"));
+                            return;
+                        }
+                        (*ctx->entry)["receiver_data"].push_back(report::Entry{
+                              {"connect_time", ctx->txp->connect_time()},
+                              {"constant_bitrate", *constant_bitrate != 0},
+                              {"delta_user_time", 0.0},
+                              {"delta_sys_time", 0.0},
+                              {"elapsed", time_elapsed},
+                              {"elapsed_target", *elapsed_target},
+                              {"engine_name", "libmeasurement_kit"},
+                              {"engine_version", MK_VERSION},
+                              {"fast_scale_down", *fast_scale_down},
+                              {"internal_address",
+                               ctx->txp->sockname().hostname},
+                              {"iteration", ctx->iteration},
+                              {"platform", mk_platform()},
+                              {"rate", rate_kbit},
+                              {"real_address", ctx->real_address},
+                              /*
+                               * Note: here we're only concerned with the amount
+                               * of useful data we received (we ignore overhead)
+                               *
+                               * This is different from the original
+                               * implementation of DASH that is part of Neubot.
+                               */
+                              {"received", length},
+                              {"remote_address", ctx->txp->peername().hostname},
+                              {"request_ticks", saved_time},
+                              {"timestamp", llround(saved_time)},
+                              {"use_fixed_rates", *use_fixed_rates},
+                              {"uuid", ctx->uuid},
+                              /*
+                               * This version indicates measurement-kit.
+                               */
+                              {"version", "0.007000000"}});
```

Comment:
Neubot's latest version number is 0.4.16.9 ("0.004016009"). Given some room for one-two more releases (if ever) plus the fact that MK is at 0.7.0, I figured to indicate there the version number of MK starting from 0.7.0.